src/client/collectors.ts

```diff
@@ -80,39 +82,39 @@ export class Collector extends HarmonyEventEmitter<CollectorEvents> {
   }
 
   /** End collecting */
-  end(): Collector {
+  end(): this {
     this.started = false
     if (this._timer !== undefined) clearTimeout(this._timer)
     return this
   }
 
   /** Reset collector and start again */
-  reset(): Collector {
+  reset(): this {
     this.collected = new Collection()
     this.collect()
     return this
   }
 
   /** Init the Collector on Client */
-  init(client: Client): Collector {
+  init(client: Client): this {
     this.client = client
     client.addCollector(this)
     return this
   }
 
   /** De initialize the Collector i.e. remove cleanly */
-  deinit(client: Client): Collector {
+  deinit(client: Client): this {
     client.removeCollector(this)
     return this
   }
 
   /** Checks we may want to perform on an extended version of Collector */
-  protected check(..._args: any[]): boolean | Promise<boolean> {
+  protected check(..._args: T): boolean | Promise<boolean> {
```

Comment:
not sure but this might give a TypeError because rest parameters should away be array's

Reply:
T has a constraint of being Array type (so tuples like `[Message]` etc)

Reply:
oh okay yeah then it is fine to just use `T`